compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1314,6 +1318,24 @@ impl Type {
         }
     }
 
+    /// Empty arrays and strings (which are arrays under the hood) are disallowed 
+    /// as input to program entry points. 
+    ///
+    /// The point of inputs to entry points is to process input data. 
+    /// Thus, passing empty arrays is pointless and adds extra complexity to the compiler
+    /// for handling them.
+    fn array_or_string_len_is_not_zero(&self) -> bool {
+        match self {
+            Type::Array(length, _) | Type::String(length) => {
+                let length = length
+                .evaluate_to_u32(Location::dummy())
+                .expect("Cannot have variable sized arrays as a parameter to main");
+                length != 0
```

Comment:
```suggestion
                let length = length
                    .evaluate_to_u32(Location::dummy())
                    .unwrap_or(0);
                length != 0
```
I think we should avoid a panic here